test/test_posture_estimator.cpp

```diff
@@ -0,0 +1,272 @@
+                    0.00001);  // Rounding due to using float
+};
+
+BOOST_AUTO_TEST_CASE(LinesAngleInAllQuads) {
+  PostProcessing::Coordinate p = {0, 0, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate q1 = {2, 2, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate q2 = {-2, 2, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate q3 = {-2, -2, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate q4 = {2, -2, PostProcessing::Trustworthy};
+  PostureEstimating::PostureEstimator e;
+  float angleQ1 = e.getLineAngle(p, q1);
+  float angleQ2 = e.getLineAngle(p, q2);
+  float angleQ3 = e.getLineAngle(p, q3);
+  float angleQ4 = e.getLineAngle(p, q4);
+
+  BOOST_CHECK_CLOSE(angleQ1, ((45.0 / 180.0) * M_PI),
+                    0.00001);  // Rounding due to using float
+  BOOST_CHECK_CLOSE(angleQ4, ((135.0 / 180.0) * M_PI),
+                    0.00001);  // Rounding due to using float
+  BOOST_CHECK_CLOSE(angleQ2, (-45.0 / 180.0) * M_PI,
+                    0.00001);  // Rounding due to using float
+  BOOST_CHECK_CLOSE(angleQ3, ((-135.0 / 180.0) * M_PI),
+                    0.00001);  // Rounding due to using float
+};
+
+BOOST_AUTO_TEST_CASE(LinesAngleSlopeZero) {
+  PostProcessing::Coordinate p = {1, 1, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate h_p = {4, 1, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate v_p = {1, 4, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate h_n = {-4, 1, PostProcessing::Trustworthy};
+  PostProcessing::Coordinate v_n = {1, -4, PostProcessing::Trustworthy};
+  PostureEstimating::PostureEstimator e;
+
+  BOOST_CHECK_CLOSE(e.getLineAngle(p, h_p), M_PI / 2,
+                    0.00001);  // Rounding due to using float
+  BOOST_CHECK_EQUAL(e.getLineAngle(p, v_p), 0);
+  BOOST_CHECK_CLOSE(e.getLineAngle(p, h_n), -M_PI / 2,
+                    0.00001);  // Rounding due to using float
+  BOOST_CHECK_CLOSE(e.getLineAngle(p, v_n), -M_PI, 0.00001);
+};
+
+BOOST_AUTO_TEST_CASE(EmptyWorkingPoseChanges) {
+  PostureEstimating::PostureEstimator e;
+  e.calculatePoseChanges();
+
+  for (int i = JointMin; i <= JointMax; i++) {
+    BOOST_CHECK_EQUAL(e.pose_changes.joints[i]->upper_angle, 0);
+    BOOST_CHECK_EQUAL(e.pose_changes.joints[i]->lower_angle, 0);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(NoDifferenceWorkingPoseChanges) {
+  PostureEstimating::PostureEstimator e;
+  PostureEstimating::destroyPose(e.ideal_pose);
+  PostureEstimating::destroyPose(e.current_pose);
+  e.ideal_pose = helper_create_pose();
+  e.current_pose = helper_create_pose();
+  e.calculatePoseChanges();
+
+  for (int i = JointMin; i <= JointMax; i++) {
+    BOOST_CHECK_EQUAL(e.pose_changes.joints[i]->upper_angle, 0);
+    BOOST_CHECK_EQUAL(e.pose_changes.joints[i]->lower_angle, 0);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(WorkingPoseChanges) {
+  PostureEstimating::PostureEstimator e;
+  PostureEstimating::destroyPose(e.ideal_pose);
+  PostureEstimating::destroyPose(e.current_pose);
```

Comment:
We seem to destroy the ideal and current_pose straight away most of the time, should we be doing something different in the constructor?

Reply:
That's just for tests. I wanted to make sure that the creating a pose and testing a pose where separated, in case there's an error in creating a pose from results.

Reply:
Makes sense, thanks for clarifying!